File: ObjcSample/ViewController.m
//
//  ViewController.m
//  ObjcSample
//
//  Created by Julian Gruber on 14/07/2017.
//  Copyright © 2017 Julian Gruber. All rights reserved.
//

#import "ViewController.h"

@import TwitterSearchKit;

#define kReuseIdentifier @"reuseIdentifier"

@interface ViewController ()
@property (nonatomic, strong) NSArray<NSDictionary *> *tweets;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Sample App";
    
    [self.tableView registerClass:UITableViewCell.class forCellReuseIdentifier:kReuseIdentifier];
    
    [TSTwitterSearchKit getTweetsWithSearchTerm:@"Radiohead" :^(NSArray<NSDictionary<NSString *,id> *> * _Nullable tweets) {
        self.tweets = tweets;
        [self.tableView reloadData];
    }];
}


#pragma mark - Table View

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    if(self.tweets != nil) {
        return self.tweets.count;
    } else {
        return 0;
    }
}

- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 45;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:kReuseIdentifier forIndexPath:indexPath];
 
    // Configure the cell...
    cell.textLabel.text = @"Tweet unavailable";
    NSDictionary *tweet = self.tweets[indexPath.row];
    if(tweet != nil) {
        id textObj = tweet[@"text"];
        if([textObj isKindOfClass:NSString.class]) {
            cell.textLabel.text = (NSString *)textObj;
        }
    }
 
    return cell;
}


@end

File: SwiftSample/ViewController.swift
//
//  ViewController.swift
//  SwiftSample
//
//  Created by Julian Gruber on 13/07/2017.
//  Copyright © 2017 Julian Gruber. All rights reserved.
//

import UIKit
import TwitterSearchKit

class ViewController: UITableViewController {

    let reuseIdentifier:String = "reuseIdentifier"
    
    var tweets:[[String:Any]]?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Sample App"
        
        self.tableView.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: self.reuseIdentifier)
        
        TwitterSearchKit.getTweets(withSearchTerm: "Radiohead") { (tweets) in
            self.tweets = tweets
            self.tableView.reloadData()
        }
    }
    
    // MARK: - Table View
    
    override func numberOfSections(in tableView: UITableView) -> Int {
        return 1
    }
    
    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.tweets?.count ?? 0
    }
    
    override func tableView(_ tableView: UITableView, heightForRowAt indexPath: IndexPath) -> CGFloat {
        return 45
    }
    
    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell = tableView.dequeueReusableCell(withIdentifier: self.reuseIdentifier, for: indexPath)
     
        // Configure the cell...
        if let tweet = self.tweets?[indexPath.row], let text = tweet["text"] as? String {
            cell.textLabel?.text = text
        } else {
            cell.textLabel?.text = "Tweet unavailable"
        }
     
        return cell
    }
    
}

